pkg/linode-bs/controllerserver.go

```diff
@@ -102,6 +102,8 @@ func (linodeCS *LinodeControllerServer) CreateVolume(ctx context.Context, req *c
 		volumeContext[LuksKeySizeAttribute] = req.Parameters[LuksKeySizeAttribute]
 	}
 
+	tags := req.Parameters[VolumeTags]
```

Comment:
Also I'm sure this is tested, but can `req.Parameters[VolumeTags]` return nil for the tags?

I don't think we check and `strings.split` is called directly later on on this tags variable that cannot be nil

Reply:
https://go.dev/ref/spec#Index_expressions

```
For a of [map type](https://go.dev/ref/spec#Map_types) M:

x's type must be [assignable](https://go.dev/ref/spec#Assignability) to the key type of M
if the map contains an entry with key x, a[x] is the map element with key x and the type of a[x] is the element type of M
if the map is nil or does not contain such an entry, a[x] is the [zero value](https://go.dev/ref/spec#The_zero_value) for the element type of M
```

I was relying on #3 in the above to guarantee a non-nil string

Reply:
i did this math when we added the luks stuff, map[string]string will return an empty string, not nil: https://go.dev/play/p/myxllIPEiqe

the other stuff like strings.Split will be fine with an empty string input.